provider/common/availabilityzones.go

```diff
@@ -148,27 +151,45 @@ func AvailabilityZoneAllocations(env ZonedEnviron, ctx context.ProviderCallConte
 var internalAvailabilityZoneAllocations = AvailabilityZoneAllocations
 
 // DistributeInstances is a common function for implement the
-// state.InstanceDistributor policy based on availability zone
-// spread.
-func DistributeInstances(env ZonedEnviron, ctx context.ProviderCallContext, candidates, group []instance.Id) ([]instance.Id, error) {
-	// Determine the best availability zones for the group.
+// state.InstanceDistributor policy based on availability zone spread.
+func DistributeInstances(
+	env ZonedEnviron, ctx context.ProviderCallContext, candidates, group []instance.Id, limitZones []string,
```

Comment:
I wonder if we've grown to the point where we would want to pass in a more generic struct containing restrictions, rather than yet-another parameter.

That can be done whenever we come back to this function. Maybe just a comment that we might like to do so?